src/org/wikipedia/Wiki.java

```diff
@@ -2251,7 +2250,8 @@ public void purge(boolean links, String... titles) throws IOException
             url.append(encode(template, false));
         }
         url.append("&titles=");
-        
+
+        String[] titlestrings = constructTitleString(url.length() + "&tlcontinue=".length(), titles, true);
```

Comment:
The length of the base URL needs to be a little longer here to accommodate the value of ``tlcontinue``.